hera_sim/visibilities/simulators.py

```diff
@@ -0,0 +1,333 @@
+                          "intensity to approximate points.")
+
+            (pos,
+             flux) = self.convert_healpix_to_point_sources(self.sky_intensity)
+
+            if self.point_source_pos is None:
+                self.point_source_pos = pos
+                self.point_source_flux = flux
+            else:
+                self.point_source_flux = \
+                    np.hstack((self.point_source_flux, flux))
+                self.point_source_pos = np.hstack((self.point_source_pos, pos))
+
+            self.sky_intensity = None
+
+    @staticmethod
+    def convert_point_sources_to_healpix(point_source_pos, point_source_flux,
+                                         nside=2**5):
+        """
+        Convert point sources to an approximate diffuse HEALPix model.
+
+        The healpix map returned is in RING scheme.
+
+        Parameters
+        ----------
+        point_source_pos : array_like
+            An array of point sources. For each source, the entries are
+            (ra, dec) [rad] (assumed to be in J2000).
+            Shape=(N_SOURCES, 2).
+        point_source_flux : array_like
+            point_source_flux : array_like, optional
+            An array of fluxes of the given point sources, per
+            frequency. Fluxes in [Jy]. Shape=(NFREQS, N_SOURCES).
+        nside : int, optional
+            HEALPix nside parameter (must be a power of 2).
+
+        Returns
+        -------
+        array_like
+            The HEALPix diffuse model. Shape=(NFREQ, NPIX).
+        """
+
+        hmap = np.zeros((len(point_source_flux), healpy.nside2npix(nside)))
+
+        # Get which pixel every point source lies in.
+        pix = healpy.ang2pix(nside, np.pi/2 - point_source_pos[:, 1],
+                             point_source_pos[:, 0])
+
+        hmap[:, pix] += point_source_flux / healpy.nside2pixarea(nside)
+
+        return hmap
+
+    @staticmethod
+    def convert_healpix_to_point_sources(hmap):
+        """
+        Convert a HEALPix map to a set of point sources.
+
+        The point sources are placed at the center of each pixel.
+
+        Parameters
+        ----------
+        hmap : array_like
+            The HEALPix map. Shape=(NFREQ, NPIX).
+
+        Returns
+        -------
+            array_like
+                The point source approximation. Positions in (ra, dec) (J2000).
+                Shape=(N_SOURCES, 2). Fluxes in [Jy]. Shape=(NFREQ, N_SOURCES).
+        """
+        nside = healpy.get_nside(hmap[0])
+        ra, dec = healpy.pix2ang(nside, np.arange(len(hmap[0])), lonlat=True)
+        flux = hmap * healpy.nside2pixarea(nside)
+        return np.array([ra*np.pi/180, dec*np.pi/180]).T, flux
+
+    def simulate(self):
+        """Perform the visibility simulation."""
+        self._write_history()
+        vis = self._simulate()
+        self.uvdata.data_array += vis
+        return vis
+
+    def _simulate(self):
```

Comment:
Make this an `abstractmethod`. In case you don't know the proper way to do so, here's a quick rundown:

1. add to your imports: `from abc import ABCMeta, abstractmethod`
2. update the class declaration to `class VisibilitySimulator(object, metaclass=ABCMeta):`
3. decorate the `_simulate` method with `@abstractmethod`

Voila! This will enforce that every subclass of `VisibilitySimulator` *must* define a `_simulate` method.

Reply:
Done (except with py2 compatibility for now).